### ISSUE TITLE
[Architecture] - Add an example where we hold a single cache in services and propagate granular model events to UI

### ISSUE THREAD

**Issue by alexjba** [status-im/status-desktop#14561]
#### Context:
--- Add here the new architecture ---

#### Goal:

The goal of this task is to establish an approach for implementing granular model updates while using a single cache in the services layer. This approach is aimed to streamline the development efforts since all of us will face this challenge in the first iteration of this new architecture. 

##### Proposal of standardised implementation
 - Develop a proposal outlining the approach we can use to update the qml layer in a granular manner while holding a single data cache in services. This proposal is something to be presented to the status dev team to streamline the development efforts for new features where the new architecture is used.
     - Showcase the integration of granular model updates into both existing and new feature. This should address common scenarios and potential challenges we may encounter. (E.g handling status-go updates received by events, handling updates requested by `get` RPCs)
     - Nice to have: bulk insert/update/remove in the model. This would boost the proxy models efficiency, but I'm not sure how difficult would be to come up with a reusable approach. Could be something to have in mind and create a new task if needed.

##### Implementation of reference example
- Implement a reference example following the proposed standardised approach. This example should showcase the practical application of granular model updates in a real-life scenario.

#### Deliverables:
1. Present this approach to the status devs
2. Reference Example Implementation

**Comment by jrainville** [comment]
@alexjba can this be closed now?

**Comment by noeliaSD** [comment]
> [@alexjba](https://github.com/alexjba) can this be closed now?

I don't think we have a full example as it's described here. I'd keep it and move it as children of [this](https://github.com/status-im/status-desktop/issues/17872)
